packages/ia-components/sandbox/details/DetailsDownloadOptions.jsx

```diff
@@ -22,37 +25,40 @@ import { I18nSpan, I18nStr, I18nIcon } from "../languages/Languages";
 
 export default class DetailsDownloadOptions extends React.Component {
   downloadableFilesDict() {
-    return this.props.files.reduce((res, af) => {
-      const format = af.metadata.format;
-      if (downloadableFormat(format)) {
-        if (!res[format]) { res[format] = []; }
-        res[format].push(af);
-      }
-      return res;
-    }, {});
+    return this.props.files
+      .filter(af => ( !this.props.disconnected || af.downloaded))
+      .reduce((res, af) => {
+        const format = af.metadata.format;
+        if (downloadableFormat(format)) {
+          if (!res[format]) { res[format] = []; }
+          res[format].push(af);
+        }
+        return res;
+      }, {});
   }
 
 
   render() {
     // Build a dictionary of file formats
-    //TODO Add the 'reachable' test in Anchor Download to this filter
-    //TODO See https://github.com/internetarchive/dweb-mirror/issues/246 for missing files issue
+    // TODO Add the 'reachable' test in Anchor Download to this filter
+    // TODO See https://github.com/internetarchive/dweb-mirror/issues/246 for missing files issue
 
     const downloadableFilesDict = this.downloadableFilesDict();
     const filesCount = this.props.files_count;
     const originalFilesCount = this.props.files.filter(f => f.metadata.source === 'original').length + 1; // Adds in Archive BitTorrent
```

Comment:
suggestion: destructure `this.props`

Reply:
Can you explain why destructing props is better than an assignment, I've seen that preference expressed in esLint, and have never seen a good explanation of why 
```
const {filesCount} = this.props;
```
is in any way better than 
```
const filesCount = this.props.filesCount
```
IMHO its neither more efficient, nor more readable nor less prone to error.

Reply:
Given that `this.props` will always exist in a React component, traversing 1 level is fine, but what happens when we go further?
Things happen, and I don't want one lil JS exception to stop the whole show. 

I think having code that actively guards against unwanted behavior just part of the good JS habits we want to uphold.

> IMHO its neither more efficient, nor more readable nor less prone to error.
Ya, sure. But we are writing ES6, let's lean into it.

Reply:
The new features in ES6 don't eliminate all the existing features of javascript.  Someone at AirBnB decided that they prefered destructuring so decided to impose it on everyone else, when (for single values) there is zero advantage AFAIK.  

I'm not sure what you mean by traversing multiple levels - if you mean for example something like 
```
const foo = this.props.bar.baz
```
Then you need to handle the case when `bar` isn't set in props, but checking that is required with destructuring as well - and in fact I wouldn't even know how to write such a two-level destructure. 

Note this is not in code I've changed in this PR, and I'm certainly not going to go back through all my code and make changes to match that particular preference for zero gains in performance, reliability, or error prone-ness.